ompi/mca/part/persist/part_persist_sendreq.h

```diff
@@ -0,0 +1,94 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2013 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2006 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2015-2017 Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2015      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * Copyright (c) 2017      Intel, Inc. All rights reserved
+ * Copyright (c) 2020-2021 Sandia National Laboratories. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef PART_PERSIST_PSENDREQ_H
+#define PART_PERSIST_PSENDREQ_H
+
+#include "ompi/mca/part/persist/part_persist_request.h"
+#include "ompi/mca/part/base/part_base_psendreq.h"
+#include "ompi/mca/part/part.h"
+#include "opal/prefetch.h"
+
+struct mca_part_persist_psend_request_t {
+    mca_part_persist_request_t req_base;
+};
+typedef struct mca_part_persist_psend_request_t mca_part_persist_psend_request_t;
+OBJ_CLASS_DECLARATION(mca_part_persist_psend_request_t);
+
+
+#define MCA_PART_PERSIST_PSEND_REQUEST_ALLOC(sendreq, comm, dst,          \
+                                           ompi_proc)                 \
+do {                                                                  \
+    sendreq = (mca_part_persist_psend_request_t*)                         \
+        opal_free_list_wait (&mca_part_base_psend_requests);          \
+    sendreq->req_base.req_type = MCA_PART_PERSIST_REQUEST_PSEND;          \
+} while(0)
+
+#define MCA_PART_PERSIST_PSEND_REQUEST_INIT( req_send,                    \
```

Comment:
nit: At this point I believe we recommend always-inline static inline vs macros. I am fine with these they way they are but it does somewhat hurt readability.

Reply:
@hjelmn The most recent commit addresses all of the requested changes except for this one. If this is a blocker I can  go back and change the macros, but my inclination is to leave them as is for now if that works.